src/util.h

```diff
@@ -494,14 +494,21 @@ class BufferValue : public MaybeStackBuffer<char> {
 // silence a compiler warning about that.
 template <typename T> inline void USE(T&&) {}
 
-// Run a function when exiting the current scope.
-struct OnScopeLeave {
-  std::function<void()> fn_;
```

Comment:
couldn't you just change this to `void(*)()` instead of reworking the entire thing?

Reply:
No, that would remove the ability to capture variables and through that make it much less useful as a shortcut